server/json-rpc/src/lib.rs

```diff
@@ -244,15 +254,25 @@ pub fn parse_response(response: Value) -> Result<(), jsonrpc_core::Error> {
 }
 
 fn subgraph_routes(name: &SubgraphName, http_port: u16, ws_port: u16) -> Value {
+    let http_base_url = EXTERNAL_HTTP_BASE_URL
+        .clone()
+        .unwrap_or_else(|| format!(":{}", http_port));
+    let ws_base_url = EXTERNAL_WS_BASE_URL
+        .clone()
+        .unwrap_or_else(|| format!(":{}", ws_port));
```

Comment:
Actually, looking at this, I'd like the env vars to override the ports altogether, so you have full control  over the base url and potentially base paths. If you serve the Graph node from behind a proxy, both http and ws might be served from the same port and different paths. This solution wouldn't work with that kind of setup.

Reply:
That's accounted for here. EXTERNAL_*_BASE_URL can be something like `http://www.example.com:8080/some/path` if needed